### PR TITLE
feat: team members management support

### LINKED ISSUE #683: Team Members
Add the ability to add and remove people from Teams

### PR DESCRIPTION
**Pull request by Prashanth-Nelli**
Closes #683

### CODE REVIEW

File: src/components/TeamMembers/TeamMembers.tsx [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+          >
+            Add Members
+          </button>
+        </div>
+      </div>
+      {Boolean(teamMembers.length) && (
+        <>
+          <h3 className="text-sm font-medium text-gray-500">
+            Team members previously added
+          </h3>
```
Comment (moshloop) [inline src/components/TeamMembers/TeamMembers.tsx:118]: I don't think we need this

```suggestion
```

File: src/components/TeamMembers/TeamMembers.tsx [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+      return;
+    }
+    fetchTeamMembers(teamId);
+  }, [teamId]);
+
+  return (
+    <div className="p-4 pb-8 space-y-2 bg-white rounded shadow">
+      <h2 className="text-dark-gray font-bold mr-3 text-xl flex items-center space-x-2">
+        Team Members
+      </h2>
```
Comment (moshloop) [inline src/components/TeamMembers/TeamMembers.tsx:93]:
```suggestion
```

File: src/components/TeamMembers/TeamMembers.tsx [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+  }
+
+  async function addTeamMembers(selectedUsers: User[]) {
+    try {
+      const { error } = await addUserToTeam(
+        teamId,
+        selectedUsers.map((item) => item.id)
+      );
+      if (!error) {
+        toastSuccess("team members added successfully");
```
Comment (moshloop) [inline src/components/TeamMembers/TeamMembers.tsx:71]: Can we make this User.name added to team successfully ?

File: src/components/SchemaResourcePage/resourceTypes.tsx [hunk 4]
```diff
@@ -19,6 +19,16 @@ export const schemaResourceTypes = [
...
     api: "incident-commander",
     icon: UserGroupIcon,
+    subNav: [
+      {
+        label: "Spec",
+        value: "spec"
+      },
+      {
+        label: "Manage Team",
+        value: "manageTeam"
```
Comment (moshloop) [inline src/components/SchemaResourcePage/resourceTypes.tsx:29]:
```suggestion
        label: "Members",
        value: "manageTeam"
```

File: src/components/TeamMembers/TeamMembers.tsx [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+          )}
+          <button
+            type="submit"
+            className="btn-primary mt-4"
+            onClick={() => setOpen(true)}
+          >
+            Add Members
+          </button>
+        </div>
+      </div>
```
Comment (moshloop) [inline src/components/TeamMembers/TeamMembers.tsx:113]: This block seems a bit repetitive - Should this use an EmptyState component (Maybe with the addition of the icon) ?

File: src/components/TeamMembers/TeamMembers.tsx (hunk 3 above)
Comment (Prashanth-Nelli) [reply src/components/TeamMembers/TeamMembers.tsx:71]: we can also add multiple users at once, in that should I include all added user names in the toast message.

**Pull request merged**